### ISSUE TITLE
Build Issue " unknown property cdvCompileSdkVersion"

### ISSUE THREAD

**Issue by Jason6Ping** [agamemnus/cordova-plugin-xapkreader#129]
Hi, 
I met build issue after I install the plugin, the error is:

"_Could not get unknown property 'cdvCompileSdkVersion' for object of type com.android.build.gradle.LibraryExtension._"

I just follow the steps mentioned in [Adding an Android APK Expansion File to a Cordova Project](https://iphonedevlog.wordpress.com/2014/12/12/adding-an-android-apk-expansion-file-to-a-cordova-project/) and also downgraded the Cordova to 6.5.

Also, I have tried change the "build-extra.gradle" file in this plugin by modifying dependencies to "_implementation project(':com.flyingsoftgames.xapkreader:library')_" but I still get same error.

Does anyone have idea on this?
Thanks.

**Comment by mv83** [comment]
I have the same problem...

**Comment by pomcanysdev** [comment]
Hey,
Any news on this or were you able to solve it? I'm getting the same error.

**Comment by stihlsicher** [comment]
Hi,

four months after the last request I ran into the same issue... 
Is there any information about solving this error?
I already took a look at the other build issues but non of these seems to fix this problem...

**Comment by agamemnus** [comment]
In the fork timeline, a few people forked to 8.0. Check if that solves your
issue.

>

**Comment by stihlsicher** [comment]
> In the fork timeline, a few people forked to 8.0. Check if that solves your issue.
> […](#)

Is there an easier way to find than checking all forks? Sorry, I am not the firm with github...

**Comment by agamemnus** [comment]
Yes, there's a timeline somewhere, although when I checked just now it was gone, or maybe I checked in the wrong place.

In any event, I did incidentally look at the forks earlier yesterday, so here's one to try:
https://github.com/devtobo/cordova-plugin-xapkreader

Let me know how it goes!

**Comment by stihlsicher** [comment]
First of all: thanks a lot for your support and linking to the version.

I have tried different combinations of cordova-android  and the plugin.
Due to some building errors ("Execution failed for task ':app:mergeDebugAssets'.
> unable to create new native thread") that occured randomly I just recreated my complete app using the newest version of all tools.  I suppose these errors were releated to adding and removing different android- platform versions.

my steps:
- create new project
- add all plugins (incl. the linked version of xapkreader)
- add platform android@latest (8.0.0)
- build
Build is terminated with error:
```
* Where:
Build file '/Users/jahn/appONE/app-v2/hamminkelnNew/platforms/android/com.flyingsoftgames.xapkreader/apphamminkeln-downloader_library/build.gradle' line: 45

* What went wrong:
A problem occurred evaluating project ':com.flyingsoftgames.xapkreader:downloader_library'.
> Could not get unknown property 'cdvCompileSdkVersion' for object of type com.android.build.gradle.LibraryExtension.

```
 😞 
As far as I get from cordova documentation cdvCompileSdkVersion is a build wide variable that should be available? Has anyone further ideas?

**Comment by erobertson42** [comment]
This issue seems to stem from Cordova using an outdated `plugin-build.gradle` file, which is used to generate the final plugin `build.gradle` configs.  If anyone is still having a problem with this, I was able to fix it using a `before_plugin_install` hook script.  See here:
https://github.com/erobertson42/cordova-plugin-xapkreader/blob/cordova-9/scripts/before_install.js

I'm also doing a couple other things in this script unrelated to this issue, so if they don't apply to you, just remove lines 11-15.

**Comment by ctfrancia** [comment]
I'm still getting this issue even after the 8.0.0 release :(
